src/components/ChecSlideoutPanel.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <transition
+    appear
+    name="panel"
+    :duration="500"
+  >
+    <div class="slideout-panel">
+      <Overlay
+        v-if="renderOverlay"
```

Comment:
I think there should always be an overlay. I assume from the designs that it would be white (maybe the colour should be configurable) with a lower opacity, then if you stack them it would just overlay the previous one and have a slightly indented left side.